src/backend/distributed/utils/datautils.c

```diff
@@ -0,0 +1,32 @@
+/*-------------------------------------------------------------------------
+ *
+ * datautils.c
+ *
+ * This file contains functions to perform useful operations on data types.
+ *
+ * Copyright (c) 2014-2016, Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "postgres.h"
+#include "c.h"
+
+#include "distributed/datautils.h"
+
+
+/*
+ * Allocates a new uint64 on the heap and copies the input to that space,
+ * returning a pointer to allocated memory. Useful for lists with 64-bit
+ * integers.
+ */
+uint64 *
+AllocateUint64(uint64 value)
+{
+	uint64 *allocatedValue = (uint64 *) palloc0(sizeof(uint64));
```

Comment:
(a) PostgreSQL doesn't have unsigned methods like that, no?

(b) I don't quite understand how this would help here… I _need_ to allocate a `uint64` in the `TopTransactionContext`, so it can be added to a list and live there in that context. If it were a smaller `int`, I could just put it in an `int` list, but there's nowhere to put a 64-bit value in a list, even it can be pass-by-value on a certain platform.

In other words, a pointer list is all I've got, so I need to point to something, which necessitates allocation. How can I circumvent allocation here? It seems less _overhead_ and more required. Am I missing something?